frontend-v2/src/components/Navbar/Navbar.tsx

```diff
@@ -0,0 +1,147 @@
+"use client";
+
+import { useState } from "react";
+import Image from "next/image";
+import logo from "../../../public/uni-lectives.svg";
+import { BookOpenIcon, PencilSquareIcon, ShieldCheckIcon, UserCircleIcon, BarsArrowDownIcon, BarsArrowUpIcon, MoonIcon, ArrowRightOnRectangleIcon } from "@heroicons/react/24/outline";
+import Tooltip from "@/components/Tooltip/Tooltip";
+
+export default function Navbar() {
+  const [doLogout, setDoLogout] = useState(false);
+  const [collapsed, setCollapsed] = useState(true);
+
+  return (
+    <>
+      {/* Collapsed View */}
+      {collapsed ? (
+        <div className="fixed flex flex-col items-center w-20 h-screen gap-4 p-4 bg-gray-50 z-50 xs:p-2 xs:w-15 xs:gap-2">
```

Comment:
i think a better way to structure the logic of this navbar would be 
- code up the expanded navbar logic
- use the collapsed ternary statement _only_ in places where components differ e.g. logout button, browse courses etc 

to reduce the repetition in components throughout this navbar as much as possible